java-sdk/ldapjdk/src/main/java/netscape/ldap/controls/LDAPPagedResultsControl.java

```diff
@@ -0,0 +1,239 @@
+
+/**
+ * Represents an LDAP v3 server control that specifies a simple pagd result
+ * manipulation, which allows your LDAP client to get entries in multiple chunks
+ * (The OID for this control is 1.2.840.113556.1.4.319).
+ * <P>
+ *
+ * To use paged search you create a "paged search" control that specifies
+ * the page size and the cookie from last search.  You include the control in a
+ * search request. When a search is performed only a page is returned to the client
+ * and a new search has to be performed to access the following elements.
+ * <P>
+ *
+ *
+ * When constructing an <CODE>LDAPPagedResultsControl</CODE> object,
+ * you can specify the following information:
+ * <P>
+ *
+ * <UL>
+ * <LI>the size of the page to be returned
+ * <LI>the cookie to keep track of the previous page. This is <CODE>null</CODE>
+ * for the first page and the server returned value for the following.
+ * </UL>
+ * <P>
+ *
+ * For example:
+ * <PRE>
+ * ...
+ *      LDAPConnection ld = new LDAPConnection();
+ * 
+ *      try {
+ *          // Connect to server.
+ *          ld.connect(3,3, hostname, portnumber, "", "" );
+ *
+ *          LDAPPagedResultsControl pagecon = new LDAPPagedResultsControl(false, 3);
+ *          // Set the search constraints to use that control.
+ *          LDAPSearchConstraints cons = new LDAPSearchConstraints();
+ *          cons.setBatchSize(1);
+ *          cons.setServerControls(pagecon);
+ * 
+ *          // Start the paged search.
+ *          String cookie = null;
+ *          int pag = 1;
+ *          do{
+ *              LDAPSearchResults res = ld.search(baseDn,
+ *                      LDAPv3.SCOPE_SUB, filter, null, false, cons);
+ * 
+ *              // Loop through the incoming results.
+ *              while (res.hasMoreElements()) {
+ *                  LDAPEntry entry = res.next();
+ *                  System.out.println("DN: " + entry.getDN());
+ *              }
+ *              for (LDAPControl c: res.getResponseControls()){
+ *                  if(c instanceof LDAPPagedResultsControl resC){
+ *                      cookie = resC.getCookie();
+ *                      System.out.println("The control for pag " + pag + " return a total or " + resC.getPageSize() + " and cookie " + resC.getCookie());
+ *                      if(cookie!=null){
+ *                          pagecon = new LDAPPagedResultsControl(false, 3, cookie);
+ *                          cons.setServerControls(pagecon);
+ *                      }
+ *                  }
+ *              }
+ *              pag++;
+ *          } while (cookie != null && !cookie.isEmpty());
+ *      } catch (Exception e) {
+ *          e.printStackTrace();
+ *      }
+ * </PRE>
+ *
+ *
+ *
+ * @see netscape.ldap.LDAPControl
+ * 
+ * @author Marco Fargetta <mfargett@redhat.com>
+ */
+public class LDAPPagedResultsControl extends LDAPControl {
+    public static final String PAGEDSEARCH  = "1.2.840.113556.1.4.319";
+    private int pageSize;
+    private String cookie;
```

Comment:
Sorry, I just looked at the RFC more closely. The cookie is defined as an octet string which usually translates to byte array in Java. Should we use byte array here instead?

Reply:
Yes, it is reasonable. I am opening an additional PR